{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/gfrosi/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (20) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>type</th>\n",
       "      <th>title</th>\n",
       "      <th>summary</th>\n",
       "      <th>overall_design</th>\n",
       "      <th>status</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>gpl</th>\n",
       "      <th>title..9</th>\n",
       "      <th>technology</th>\n",
       "      <th>...</th>\n",
       "      <th>gsm</th>\n",
       "      <th>title..13</th>\n",
       "      <th>type..14</th>\n",
       "      <th>gpl..15</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>series_id</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>characteristics_ch2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM263933</td>\n",
       "      <td>CD4+ Tcell Resting M0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, resting</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM263934</td>\n",
       "      <td>CD4+ Tcell Resting T0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, resting</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM263935</td>\n",
       "      <td>CD4+ Tcell Activated M18</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, activated using anti-CD3/2...</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM263936</td>\n",
       "      <td>CD4+ Tcell Activated T18</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, activated using anti-CD3/2...</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE11672</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Genomic location data for ectopic p73 in H1299...</td>\n",
       "      <td>The integral role of p53 in tumor suppression ...</td>\n",
       "      <td>H1299 lung carcinoma cells were transduced wit...</td>\n",
       "      <td>Public on Aug 06 2008</td>\n",
       "      <td>18678646.0</td>\n",
       "      <td>GPL6931</td>\n",
       "      <td>Human H1299 cells p73 ChIP-seq</td>\n",
       "      <td>other</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM296634</td>\n",
       "      <td>ectopic p73 binding sites</td>\n",
       "      <td>genomic</td>\n",
       "      <td>GPL6931</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE11672,GSE11673</td>\n",
       "      <td>H1299 cells, p73 IP sample</td>\n",
       "      <td>cancer cell line</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69859</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Illumina NextSeq 500 (Homo sapiens)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM3044958</td>\n",
       "      <td>RUNX1_ChIP_MT_CB</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>Cord blood CD34 cells</td>\n",
       "      <td>mutation status: RUNX1mut transduced;\\tharvest...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69860</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Illumina NextSeq 500 (Homo sapiens)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM3044959</td>\n",
       "      <td>RUNX1_ChIP_d31_CTR_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: control;\\tharvest time: day 3...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69861</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Illumina NextSeq 500 (Homo sapiens)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM3044960</td>\n",
       "      <td>RUNX1_ChIP_d31_DOX_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: DOX-induced RUNX1mut;\\tharves...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69862</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Illumina NextSeq 500 (Homo sapiens)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM3044961</td>\n",
       "      <td>RUNX1_ChIP_d34_CTR_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: control;\\tharvest time: day 3...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69863</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Illumina NextSeq 500 (Homo sapiens)</td>\n",
       "      <td>high-throughput sequencing</td>\n",
       "      <td>...</td>\n",
       "      <td>GSM3044962</td>\n",
       "      <td>RUNX1_ChIP_d34_DOX_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: DOX-induced RUNX1mut;\\tharves...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69864 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gse                                               type  \\\n",
       "0       GSE10437                      Expression profiling by array   \n",
       "1       GSE10437                      Expression profiling by array   \n",
       "2       GSE10437                      Expression profiling by array   \n",
       "3       GSE10437                      Expression profiling by array   \n",
       "4       GSE11672  Genome binding/occupancy profiling by high thr...   \n",
       "...          ...                                                ...   \n",
       "69859  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "69860  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "69861  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "69862  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "69863  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Dynamic regulation of nucleosome positioning i...   \n",
       "1      Dynamic regulation of nucleosome positioning i...   \n",
       "2      Dynamic regulation of nucleosome positioning i...   \n",
       "3      Dynamic regulation of nucleosome positioning i...   \n",
       "4      Genomic location data for ectopic p73 in H1299...   \n",
       "...                                                  ...   \n",
       "69859  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "69860  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "69861  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "69862  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "69863  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "\n",
       "                                                 summary  \\\n",
       "0      The positioning of nucleosomes with respect to...   \n",
       "1      The positioning of nucleosomes with respect to...   \n",
       "2      The positioning of nucleosomes with respect to...   \n",
       "3      The positioning of nucleosomes with respect to...   \n",
       "4      The integral role of p53 in tumor suppression ...   \n",
       "...                                                  ...   \n",
       "69859  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "69860  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "69861  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "69862  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "69863  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "\n",
       "                                          overall_design  \\\n",
       "0      Gene expression data from resting  (M0 and T0)...   \n",
       "1      Gene expression data from resting  (M0 and T0)...   \n",
       "2      Gene expression data from resting  (M0 and T0)...   \n",
       "3      Gene expression data from resting  (M0 and T0)...   \n",
       "4      H1299 lung carcinoma cells were transduced wit...   \n",
       "...                                                  ...   \n",
       "69859  A total of six samples were subject to ChIP-Se...   \n",
       "69860  A total of six samples were subject to ChIP-Se...   \n",
       "69861  A total of six samples were subject to ChIP-Se...   \n",
       "69862  A total of six samples were subject to ChIP-Se...   \n",
       "69863  A total of six samples were subject to ChIP-Se...   \n",
       "\n",
       "                      status   pubmed_id       gpl  \\\n",
       "0      Public on Feb 09 2008  18329373.0    GPL570   \n",
       "1      Public on Feb 09 2008  18329373.0    GPL570   \n",
       "2      Public on Feb 09 2008  18329373.0    GPL570   \n",
       "3      Public on Feb 09 2008  18329373.0    GPL570   \n",
       "4      Public on Aug 06 2008  18678646.0   GPL6931   \n",
       "...                      ...         ...       ...   \n",
       "69859  Public on Feb 06 2019  30709863.0  GPL18573   \n",
       "69860  Public on Feb 06 2019  30709863.0  GPL18573   \n",
       "69861  Public on Feb 06 2019  30709863.0  GPL18573   \n",
       "69862  Public on Feb 06 2019  30709863.0  GPL18573   \n",
       "69863  Public on Feb 06 2019  30709863.0  GPL18573   \n",
       "\n",
       "                                                title..9  \\\n",
       "0      [HG-U133_Plus_2] Affymetrix Human Genome U133 ...   \n",
       "1      [HG-U133_Plus_2] Affymetrix Human Genome U133 ...   \n",
       "2      [HG-U133_Plus_2] Affymetrix Human Genome U133 ...   \n",
       "3      [HG-U133_Plus_2] Affymetrix Human Genome U133 ...   \n",
       "4                         Human H1299 cells p73 ChIP-seq   \n",
       "...                                                  ...   \n",
       "69859                Illumina NextSeq 500 (Homo sapiens)   \n",
       "69860                Illumina NextSeq 500 (Homo sapiens)   \n",
       "69861                Illumina NextSeq 500 (Homo sapiens)   \n",
       "69862                Illumina NextSeq 500 (Homo sapiens)   \n",
       "69863                Illumina NextSeq 500 (Homo sapiens)   \n",
       "\n",
       "                       technology  ...         gsm                  title..13  \\\n",
       "0         in situ oligonucleotide  ...   GSM263933      CD4+ Tcell Resting M0   \n",
       "1         in situ oligonucleotide  ...   GSM263934      CD4+ Tcell Resting T0   \n",
       "2         in situ oligonucleotide  ...   GSM263935   CD4+ Tcell Activated M18   \n",
       "3         in situ oligonucleotide  ...   GSM263936   CD4+ Tcell Activated T18   \n",
       "4                           other  ...   GSM296634  ectopic p73 binding sites   \n",
       "...                           ...  ...         ...                        ...   \n",
       "69859  high-throughput sequencing  ...  GSM3044958           RUNX1_ChIP_MT_CB   \n",
       "69860  high-throughput sequencing  ...  GSM3044959    RUNX1_ChIP_d31_CTR_iPSC   \n",
       "69861  high-throughput sequencing  ...  GSM3044960    RUNX1_ChIP_d31_DOX_iPSC   \n",
       "69862  high-throughput sequencing  ...  GSM3044961    RUNX1_ChIP_d34_CTR_iPSC   \n",
       "69863  high-throughput sequencing  ...  GSM3044962    RUNX1_ChIP_d34_DOX_iPSC   \n",
       "\n",
       "      type..14   gpl..15  organism_ch1            series_id  \\\n",
       "0          RNA    GPL570  Homo sapiens             GSE10437   \n",
       "1          RNA    GPL570  Homo sapiens             GSE10437   \n",
       "2          RNA    GPL570  Homo sapiens             GSE10437   \n",
       "3          RNA    GPL570  Homo sapiens             GSE10437   \n",
       "4      genomic   GPL6931  Homo sapiens    GSE11672,GSE11673   \n",
       "...        ...       ...           ...                  ...   \n",
       "69859      SRA  GPL18573  Homo sapiens  GSE111917,GSE111919   \n",
       "69860      SRA  GPL18573  Homo sapiens  GSE111917,GSE111919   \n",
       "69861      SRA  GPL18573  Homo sapiens  GSE111917,GSE111919   \n",
       "69862      SRA  GPL18573  Homo sapiens  GSE111917,GSE111919   \n",
       "69863      SRA  GPL18573  Homo sapiens  GSE111917,GSE111919   \n",
       "\n",
       "                                         source_name_ch1  \\\n",
       "0                            Human CD4+ T cells, resting   \n",
       "1                            Human CD4+ T cells, resting   \n",
       "2      Human CD4+ T cells, activated using anti-CD3/2...   \n",
       "3      Human CD4+ T cells, activated using anti-CD3/2...   \n",
       "4                             H1299 cells, p73 IP sample   \n",
       "...                                                  ...   \n",
       "69859                              Cord blood CD34 cells   \n",
       "69860                                               iPSC   \n",
       "69861                                               iPSC   \n",
       "69862                                               iPSC   \n",
       "69863                                               iPSC   \n",
       "\n",
       "                                     characteristics_ch1  \\\n",
       "0                                     Human CD4+ T cells   \n",
       "1                                     Human CD4+ T cells   \n",
       "2                                     Human CD4+ T cells   \n",
       "3                                     Human CD4+ T cells   \n",
       "4                                       cancer cell line   \n",
       "...                                                  ...   \n",
       "69859  mutation status: RUNX1mut transduced;\\tharvest...   \n",
       "69860  mutation status: control;\\tharvest time: day 3...   \n",
       "69861  mutation status: DOX-induced RUNX1mut;\\tharves...   \n",
       "69862  mutation status: control;\\tharvest time: day 3...   \n",
       "69863  mutation status: DOX-induced RUNX1mut;\\tharves...   \n",
       "\n",
       "                                      supplementary_file characteristics_ch2  \n",
       "0      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "1      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "2      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "3      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "4                                                    NaN                 NaN  \n",
       "...                                                  ...                 ...  \n",
       "69859                                                NaN                 NaN  \n",
       "69860                                                NaN                 NaN  \n",
       "69861                                                NaN                 NaN  \n",
       "69862                                                NaN                 NaN  \n",
       "69863                                                NaN                 NaN  \n",
       "\n",
       "[69864 rows x 21 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check the table with all fields (GPL, GSE and GSM) from geometadb related to Chip-seq and filtered to Homo sapiens\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "pd.set_option('display.max_rows', 20)\n",
    "df1 = pd.read_csv('/Users/gfrosi/Documents/Frosi_lab_jacques/XML_TEST/HS_df_chipseq_to_work_1.csv', sep=',')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gse                     4721\n",
       "type                      84\n",
       "title                   4702\n",
       "summary                 3189\n",
       "overall_design          4447\n",
       "                       ...  \n",
       "series_id               4537\n",
       "source_name_ch1         9515\n",
       "characteristics_ch1    28375\n",
       "supplementary_file      2261\n",
       "characteristics_ch2      108\n",
       "Length: 21, dtype: int64"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4721\n"
     ]
    }
   ],
   "source": [
    "#put the unique GSE into a list\n",
    "\n",
    "gse_list_GEOdb = df1['gse'].unique().tolist() \n",
    "print(len(gse_list_GEOdb)) #ok\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>gsm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE100009</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE100040</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE100074</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE100099</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE100133</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4716</th>\n",
       "      <td>GSE99905</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4717</th>\n",
       "      <td>GSE99937</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4718</th>\n",
       "      <td>GSE99943</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4719</th>\n",
       "      <td>GSE99950</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4720</th>\n",
       "      <td>GSE99998</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4721 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gse  gsm\n",
       "0     GSE100009   37\n",
       "1     GSE100040   68\n",
       "2     GSE100074   12\n",
       "3     GSE100099   90\n",
       "4     GSE100133   32\n",
       "...         ...  ...\n",
       "4716   GSE99905   15\n",
       "4717   GSE99937   28\n",
       "4718   GSE99943    2\n",
       "4719   GSE99950    1\n",
       "4720   GSE99998    5\n",
       "\n",
       "[4721 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#groupby and check some GSE manually\n",
    "\n",
    "df1_grouped = df1.groupby(['gse'])['gsm'].count().reset_index()\n",
    "df1_grouped\n",
    "\n",
    "\n",
    "\n",
    "#Example = GSE100009 has 37 samples (ok), where 17 are related to chipseq and 20 are related with RNA seq. \n",
    "##GSE100040 is inflated here (after, see why). This series contains 34 GSM (here it is with double of GSM)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Let's try with our work table no GPL\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/gfrosi/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>type</th>\n",
       "      <th>title</th>\n",
       "      <th>summary</th>\n",
       "      <th>overall_design</th>\n",
       "      <th>status</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>gsm</th>\n",
       "      <th>title..9</th>\n",
       "      <th>type..10</th>\n",
       "      <th>gpl</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>series_id</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>characteristics_ch2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GSM263933</td>\n",
       "      <td>CD4+ Tcell Resting M0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, resting</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GSM263934</td>\n",
       "      <td>CD4+ Tcell Resting T0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, resting</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GSM263935</td>\n",
       "      <td>CD4+ Tcell Activated M18</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, activated using anti-CD3/2...</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Expression profiling by array</td>\n",
       "      <td>Dynamic regulation of nucleosome positioning i...</td>\n",
       "      <td>The positioning of nucleosomes with respect to...</td>\n",
       "      <td>Gene expression data from resting  (M0 and T0)...</td>\n",
       "      <td>Public on Feb 09 2008</td>\n",
       "      <td>18329373.0</td>\n",
       "      <td>GSM263936</td>\n",
       "      <td>CD4+ Tcell Activated T18</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL570</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE10437</td>\n",
       "      <td>Human CD4+ T cells, activated using anti-CD3/2...</td>\n",
       "      <td>Human CD4+ T cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE11892</td>\n",
       "      <td>Expression profiling by array;\\tExpression pro...</td>\n",
       "      <td>A Global View of Gene Activity and Alternative...</td>\n",
       "      <td>Transcriptome of HEK and B cells were analyzed...</td>\n",
       "      <td>For HEK and B cell, samples were hybridized as...</td>\n",
       "      <td>Public on Jul 03 2008</td>\n",
       "      <td>18599741.0</td>\n",
       "      <td>GSM300493</td>\n",
       "      <td>Transcriptome of HEK293T cells</td>\n",
       "      <td>RNA</td>\n",
       "      <td>GPL6104</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE11892</td>\n",
       "      <td>HEK293T cells</td>\n",
       "      <td>Human embryonic kidney (HEK293T) cells</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM300n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47120</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GSM3044958</td>\n",
       "      <td>RUNX1_ChIP_MT_CB</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>Cord blood CD34 cells</td>\n",
       "      <td>mutation status: RUNX1mut transduced;\\tharvest...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47121</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GSM3044959</td>\n",
       "      <td>RUNX1_ChIP_d31_CTR_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: control;\\tharvest time: day 3...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47122</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GSM3044960</td>\n",
       "      <td>RUNX1_ChIP_d31_DOX_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: DOX-induced RUNX1mut;\\tharves...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47123</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GSM3044961</td>\n",
       "      <td>RUNX1_ChIP_d34_CTR_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: control;\\tharvest time: day 3...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47124</th>\n",
       "      <td>GSE111917</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>RUNX1 mutations lead to a myeloid differentiat...</td>\n",
       "      <td>Mutations in the RUNX1 gene (RUNX1mut) have be...</td>\n",
       "      <td>A total of six samples were subject to ChIP-Se...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>30709863.0</td>\n",
       "      <td>GSM3044962</td>\n",
       "      <td>RUNX1_ChIP_d34_DOX_iPSC</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE111917,GSE111919</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>mutation status: DOX-induced RUNX1mut;\\tharves...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47125 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gse                                               type  \\\n",
       "0       GSE10437                      Expression profiling by array   \n",
       "1       GSE10437                      Expression profiling by array   \n",
       "2       GSE10437                      Expression profiling by array   \n",
       "3       GSE10437                      Expression profiling by array   \n",
       "4       GSE11892  Expression profiling by array;\\tExpression pro...   \n",
       "...          ...                                                ...   \n",
       "47120  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "47121  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "47122  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "47123  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "47124  GSE111917  Genome binding/occupancy profiling by high thr...   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Dynamic regulation of nucleosome positioning i...   \n",
       "1      Dynamic regulation of nucleosome positioning i...   \n",
       "2      Dynamic regulation of nucleosome positioning i...   \n",
       "3      Dynamic regulation of nucleosome positioning i...   \n",
       "4      A Global View of Gene Activity and Alternative...   \n",
       "...                                                  ...   \n",
       "47120  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "47121  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "47122  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "47123  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "47124  RUNX1 mutations lead to a myeloid differentiat...   \n",
       "\n",
       "                                                 summary  \\\n",
       "0      The positioning of nucleosomes with respect to...   \n",
       "1      The positioning of nucleosomes with respect to...   \n",
       "2      The positioning of nucleosomes with respect to...   \n",
       "3      The positioning of nucleosomes with respect to...   \n",
       "4      Transcriptome of HEK and B cells were analyzed...   \n",
       "...                                                  ...   \n",
       "47120  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "47121  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "47122  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "47123  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "47124  Mutations in the RUNX1 gene (RUNX1mut) have be...   \n",
       "\n",
       "                                          overall_design  \\\n",
       "0      Gene expression data from resting  (M0 and T0)...   \n",
       "1      Gene expression data from resting  (M0 and T0)...   \n",
       "2      Gene expression data from resting  (M0 and T0)...   \n",
       "3      Gene expression data from resting  (M0 and T0)...   \n",
       "4      For HEK and B cell, samples were hybridized as...   \n",
       "...                                                  ...   \n",
       "47120  A total of six samples were subject to ChIP-Se...   \n",
       "47121  A total of six samples were subject to ChIP-Se...   \n",
       "47122  A total of six samples were subject to ChIP-Se...   \n",
       "47123  A total of six samples were subject to ChIP-Se...   \n",
       "47124  A total of six samples were subject to ChIP-Se...   \n",
       "\n",
       "                      status   pubmed_id         gsm  \\\n",
       "0      Public on Feb 09 2008  18329373.0   GSM263933   \n",
       "1      Public on Feb 09 2008  18329373.0   GSM263934   \n",
       "2      Public on Feb 09 2008  18329373.0   GSM263935   \n",
       "3      Public on Feb 09 2008  18329373.0   GSM263936   \n",
       "4      Public on Jul 03 2008  18599741.0   GSM300493   \n",
       "...                      ...         ...         ...   \n",
       "47120  Public on Feb 06 2019  30709863.0  GSM3044958   \n",
       "47121  Public on Feb 06 2019  30709863.0  GSM3044959   \n",
       "47122  Public on Feb 06 2019  30709863.0  GSM3044960   \n",
       "47123  Public on Feb 06 2019  30709863.0  GSM3044961   \n",
       "47124  Public on Feb 06 2019  30709863.0  GSM3044962   \n",
       "\n",
       "                             title..9 type..10       gpl  organism_ch1  \\\n",
       "0               CD4+ Tcell Resting M0      RNA    GPL570  Homo sapiens   \n",
       "1               CD4+ Tcell Resting T0      RNA    GPL570  Homo sapiens   \n",
       "2            CD4+ Tcell Activated M18      RNA    GPL570  Homo sapiens   \n",
       "3            CD4+ Tcell Activated T18      RNA    GPL570  Homo sapiens   \n",
       "4      Transcriptome of HEK293T cells      RNA   GPL6104  Homo sapiens   \n",
       "...                               ...      ...       ...           ...   \n",
       "47120                RUNX1_ChIP_MT_CB      SRA  GPL18573  Homo sapiens   \n",
       "47121         RUNX1_ChIP_d31_CTR_iPSC      SRA  GPL18573  Homo sapiens   \n",
       "47122         RUNX1_ChIP_d31_DOX_iPSC      SRA  GPL18573  Homo sapiens   \n",
       "47123         RUNX1_ChIP_d34_CTR_iPSC      SRA  GPL18573  Homo sapiens   \n",
       "47124         RUNX1_ChIP_d34_DOX_iPSC      SRA  GPL18573  Homo sapiens   \n",
       "\n",
       "                 series_id                                    source_name_ch1  \\\n",
       "0                 GSE10437                        Human CD4+ T cells, resting   \n",
       "1                 GSE10437                        Human CD4+ T cells, resting   \n",
       "2                 GSE10437  Human CD4+ T cells, activated using anti-CD3/2...   \n",
       "3                 GSE10437  Human CD4+ T cells, activated using anti-CD3/2...   \n",
       "4                 GSE11892                                      HEK293T cells   \n",
       "...                    ...                                                ...   \n",
       "47120  GSE111917,GSE111919                              Cord blood CD34 cells   \n",
       "47121  GSE111917,GSE111919                                               iPSC   \n",
       "47122  GSE111917,GSE111919                                               iPSC   \n",
       "47123  GSE111917,GSE111919                                               iPSC   \n",
       "47124  GSE111917,GSE111919                                               iPSC   \n",
       "\n",
       "                                     characteristics_ch1  \\\n",
       "0                                     Human CD4+ T cells   \n",
       "1                                     Human CD4+ T cells   \n",
       "2                                     Human CD4+ T cells   \n",
       "3                                     Human CD4+ T cells   \n",
       "4                 Human embryonic kidney (HEK293T) cells   \n",
       "...                                                  ...   \n",
       "47120  mutation status: RUNX1mut transduced;\\tharvest...   \n",
       "47121  mutation status: control;\\tharvest time: day 3...   \n",
       "47122  mutation status: DOX-induced RUNX1mut;\\tharves...   \n",
       "47123  mutation status: control;\\tharvest time: day 3...   \n",
       "47124  mutation status: DOX-induced RUNX1mut;\\tharves...   \n",
       "\n",
       "                                      supplementary_file characteristics_ch2  \n",
       "0      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "1      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "2      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "3      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM263n...                 NaN  \n",
       "4      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM300n...                 NaN  \n",
       "...                                                  ...                 ...  \n",
       "47120                                                NaN                 NaN  \n",
       "47121                                                NaN                 NaN  \n",
       "47122                                                NaN                 NaN  \n",
       "47123                                                NaN                 NaN  \n",
       "47124                                                NaN                 NaN  \n",
       "\n",
       "[47125 rows x 17 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd.read_csv('/Users/gfrosi/Documents/Frosi_lab_jacques/XML_TEST/HS_df_chipseq_to_work_no_gpl.csv', sep=',')\n",
    "df2\n",
    "\n",
    "\n",
    "#47125 GSM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df2.nunique() same GSE count\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>gsm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE100009</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE100040</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE100074</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE100099</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE100133</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4716</th>\n",
       "      <td>GSE99905</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4717</th>\n",
       "      <td>GSE99937</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4718</th>\n",
       "      <td>GSE99943</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4719</th>\n",
       "      <td>GSE99950</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4720</th>\n",
       "      <td>GSE99998</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4721 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gse  gsm\n",
       "0     GSE100009   37\n",
       "1     GSE100040   34\n",
       "2     GSE100074   12\n",
       "3     GSE100099   90\n",
       "4     GSE100133   32\n",
       "...         ...  ...\n",
       "4716   GSE99905   15\n",
       "4717   GSE99937   28\n",
       "4718   GSE99943    2\n",
       "4719   GSE99950    1\n",
       "4720   GSE99998    5\n",
       "\n",
       "[4721 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#More safely work with this table\n",
    "\n",
    "df2_grouped = df2.groupby(['gse'])['gsm'].count().reset_index()\n",
    "df2_grouped\n",
    "\n",
    "\n",
    "#Example = GSE100009 has 37 samples (ok), where 17 are related to chipseq and 20 are related with RNA seq\n",
    "#Here, the GSE10040 is correct. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>type</th>\n",
       "      <th>title</th>\n",
       "      <th>summary</th>\n",
       "      <th>overall_design</th>\n",
       "      <th>status</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>gsm</th>\n",
       "      <th>title..9</th>\n",
       "      <th>type..10</th>\n",
       "      <th>gpl</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>series_id</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>characteristics_ch2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE12781</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>PolII transcription factor in Human HeLa S3</td>\n",
       "      <td>We report the results of chromatin immunopreci...</td>\n",
       "      <td>Examination of the PolII transcription factor ...</td>\n",
       "      <td>Public on Sep 16 2008</td>\n",
       "      <td>19122651.0</td>\n",
       "      <td>GSM320734</td>\n",
       "      <td>PolII_ChIP-Seq in HeLa S3</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE12781,GSE12783</td>\n",
       "      <td>Chromatin IP against Pol II</td>\n",
       "      <td>HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE12782</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>STAT1 transcription factor in Human HeLa S3</td>\n",
       "      <td>We report the results of chromatin immunopreci...</td>\n",
       "      <td>Examination of the STAT1 transcription factor ...</td>\n",
       "      <td>Public on Sep 16 2008</td>\n",
       "      <td>19122651.0</td>\n",
       "      <td>GSM320736</td>\n",
       "      <td>IFN_gamma_STAT1_ChIP-Seq in HeLa S3</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE12782,GSE12783</td>\n",
       "      <td>Chromatin IP against STAT1 in IFN gamma stimul...</td>\n",
       "      <td>HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE12783</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Genome-wide maps of transcription factor bindi...</td>\n",
       "      <td>For data usage terms and conditions, please re...</td>\n",
       "      <td>Examination of 2 different transcription facto...</td>\n",
       "      <td>Public on Sep 16 2008</td>\n",
       "      <td>19122651.0</td>\n",
       "      <td>GSM320734</td>\n",
       "      <td>PolII_ChIP-Seq in HeLa S3</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE12781,GSE12783</td>\n",
       "      <td>Chromatin IP against Pol II</td>\n",
       "      <td>HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE12783</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Genome-wide maps of transcription factor bindi...</td>\n",
       "      <td>For data usage terms and conditions, please re...</td>\n",
       "      <td>Examination of 2 different transcription facto...</td>\n",
       "      <td>Public on Sep 16 2008</td>\n",
       "      <td>19122651.0</td>\n",
       "      <td>GSM320736</td>\n",
       "      <td>IFN_gamma_STAT1_ChIP-Seq in HeLa S3</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE12782,GSE12783</td>\n",
       "      <td>Chromatin IP against STAT1 in IFN gamma stimul...</td>\n",
       "      <td>HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE13008</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Genome-wide mapping of RNA Polymerase II bindi...</td>\n",
       "      <td>Protein coding genes and some of the non-codin...</td>\n",
       "      <td>Map binding sites of RNA Polymerase II in the ...</td>\n",
       "      <td>Public on Oct 02 2008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM325933</td>\n",
       "      <td>PolII_ChIP-Seq in K562</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE13008</td>\n",
       "      <td>Chromatin IP against Pol II</td>\n",
       "      <td>K562 cells</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13162</th>\n",
       "      <td>GSE106146</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Histone acetyltransferase p300 induces de novo...</td>\n",
       "      <td>Cellular senescence is a state of stable proli...</td>\n",
       "      <td>This study compares genome-wide ChIP-seq profi...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM2830456</td>\n",
       "      <td>p300 ChIP-seq from Senescent Fibroblasts, Repl...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE105937,GSE106146</td>\n",
       "      <td>p300 ChIP-seq from Senescent Fibroblasts</td>\n",
       "      <td>cell line: IMR90;\\tcell type: Lung Fibroblasts...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13163</th>\n",
       "      <td>GSE106146</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Histone acetyltransferase p300 induces de novo...</td>\n",
       "      <td>Cellular senescence is a state of stable proli...</td>\n",
       "      <td>This study compares genome-wide ChIP-seq profi...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM2830457</td>\n",
       "      <td>IgG ChIP-seq from Proliferating Fibroblasts, R...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE105937,GSE106146</td>\n",
       "      <td>IgG ChIP-seq from Proliferating Fibroblasts</td>\n",
       "      <td>cell line: IMR90;\\tcell type: Lung Fibroblasts...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13164</th>\n",
       "      <td>GSE106146</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Histone acetyltransferase p300 induces de novo...</td>\n",
       "      <td>Cellular senescence is a state of stable proli...</td>\n",
       "      <td>This study compares genome-wide ChIP-seq profi...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM2830458</td>\n",
       "      <td>IgG ChIP-seq from Proliferating Fibroblasts, R...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE105937,GSE106146</td>\n",
       "      <td>IgG ChIP-seq from Proliferating Fibroblasts</td>\n",
       "      <td>cell line: IMR90;\\tcell type: Lung Fibroblasts...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13165</th>\n",
       "      <td>GSE106146</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Histone acetyltransferase p300 induces de novo...</td>\n",
       "      <td>Cellular senescence is a state of stable proli...</td>\n",
       "      <td>This study compares genome-wide ChIP-seq profi...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM2830459</td>\n",
       "      <td>IgG ChIP-seq from Senescent Fibroblasts, Repli...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE105937,GSE106146</td>\n",
       "      <td>IgG ChIP-seq from Senescent Fibroblasts</td>\n",
       "      <td>cell line: IMR90;\\tcell type: Lung Fibroblasts...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13166</th>\n",
       "      <td>GSE106146</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>Histone acetyltransferase p300 induces de novo...</td>\n",
       "      <td>Cellular senescence is a state of stable proli...</td>\n",
       "      <td>This study compares genome-wide ChIP-seq profi...</td>\n",
       "      <td>Public on Feb 06 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM2830460</td>\n",
       "      <td>IgG ChIP-seq from Senescent Fibroblasts, Repli...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL18573</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE105937,GSE106146</td>\n",
       "      <td>IgG ChIP-seq from Senescent Fibroblasts</td>\n",
       "      <td>cell line: IMR90;\\tcell type: Lung Fibroblasts...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13167 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gse                                               type  \\\n",
       "0       GSE12781  Genome binding/occupancy profiling by high thr...   \n",
       "1       GSE12782  Genome binding/occupancy profiling by high thr...   \n",
       "2       GSE12783  Genome binding/occupancy profiling by high thr...   \n",
       "3       GSE12783  Genome binding/occupancy profiling by high thr...   \n",
       "4       GSE13008  Genome binding/occupancy profiling by high thr...   \n",
       "...          ...                                                ...   \n",
       "13162  GSE106146  Genome binding/occupancy profiling by high thr...   \n",
       "13163  GSE106146  Genome binding/occupancy profiling by high thr...   \n",
       "13164  GSE106146  Genome binding/occupancy profiling by high thr...   \n",
       "13165  GSE106146  Genome binding/occupancy profiling by high thr...   \n",
       "13166  GSE106146  Genome binding/occupancy profiling by high thr...   \n",
       "\n",
       "                                                   title  \\\n",
       "0            PolII transcription factor in Human HeLa S3   \n",
       "1            STAT1 transcription factor in Human HeLa S3   \n",
       "2      Genome-wide maps of transcription factor bindi...   \n",
       "3      Genome-wide maps of transcription factor bindi...   \n",
       "4      Genome-wide mapping of RNA Polymerase II bindi...   \n",
       "...                                                  ...   \n",
       "13162  Histone acetyltransferase p300 induces de novo...   \n",
       "13163  Histone acetyltransferase p300 induces de novo...   \n",
       "13164  Histone acetyltransferase p300 induces de novo...   \n",
       "13165  Histone acetyltransferase p300 induces de novo...   \n",
       "13166  Histone acetyltransferase p300 induces de novo...   \n",
       "\n",
       "                                                 summary  \\\n",
       "0      We report the results of chromatin immunopreci...   \n",
       "1      We report the results of chromatin immunopreci...   \n",
       "2      For data usage terms and conditions, please re...   \n",
       "3      For data usage terms and conditions, please re...   \n",
       "4      Protein coding genes and some of the non-codin...   \n",
       "...                                                  ...   \n",
       "13162  Cellular senescence is a state of stable proli...   \n",
       "13163  Cellular senescence is a state of stable proli...   \n",
       "13164  Cellular senescence is a state of stable proli...   \n",
       "13165  Cellular senescence is a state of stable proli...   \n",
       "13166  Cellular senescence is a state of stable proli...   \n",
       "\n",
       "                                          overall_design  \\\n",
       "0      Examination of the PolII transcription factor ...   \n",
       "1      Examination of the STAT1 transcription factor ...   \n",
       "2      Examination of 2 different transcription facto...   \n",
       "3      Examination of 2 different transcription facto...   \n",
       "4      Map binding sites of RNA Polymerase II in the ...   \n",
       "...                                                  ...   \n",
       "13162  This study compares genome-wide ChIP-seq profi...   \n",
       "13163  This study compares genome-wide ChIP-seq profi...   \n",
       "13164  This study compares genome-wide ChIP-seq profi...   \n",
       "13165  This study compares genome-wide ChIP-seq profi...   \n",
       "13166  This study compares genome-wide ChIP-seq profi...   \n",
       "\n",
       "                      status   pubmed_id         gsm  \\\n",
       "0      Public on Sep 16 2008  19122651.0   GSM320734   \n",
       "1      Public on Sep 16 2008  19122651.0   GSM320736   \n",
       "2      Public on Sep 16 2008  19122651.0   GSM320734   \n",
       "3      Public on Sep 16 2008  19122651.0   GSM320736   \n",
       "4      Public on Oct 02 2008         NaN   GSM325933   \n",
       "...                      ...         ...         ...   \n",
       "13162  Public on Feb 06 2019         NaN  GSM2830456   \n",
       "13163  Public on Feb 06 2019         NaN  GSM2830457   \n",
       "13164  Public on Feb 06 2019         NaN  GSM2830458   \n",
       "13165  Public on Feb 06 2019         NaN  GSM2830459   \n",
       "13166  Public on Feb 06 2019         NaN  GSM2830460   \n",
       "\n",
       "                                                title..9 type..10       gpl  \\\n",
       "0                              PolII_ChIP-Seq in HeLa S3      SRA   GPL9115   \n",
       "1                    IFN_gamma_STAT1_ChIP-Seq in HeLa S3      SRA   GPL9115   \n",
       "2                              PolII_ChIP-Seq in HeLa S3      SRA   GPL9115   \n",
       "3                    IFN_gamma_STAT1_ChIP-Seq in HeLa S3      SRA   GPL9115   \n",
       "4                                 PolII_ChIP-Seq in K562      SRA   GPL9115   \n",
       "...                                                  ...      ...       ...   \n",
       "13162  p300 ChIP-seq from Senescent Fibroblasts, Repl...      SRA  GPL18573   \n",
       "13163  IgG ChIP-seq from Proliferating Fibroblasts, R...      SRA  GPL18573   \n",
       "13164  IgG ChIP-seq from Proliferating Fibroblasts, R...      SRA  GPL18573   \n",
       "13165  IgG ChIP-seq from Senescent Fibroblasts, Repli...      SRA  GPL18573   \n",
       "13166  IgG ChIP-seq from Senescent Fibroblasts, Repli...      SRA  GPL18573   \n",
       "\n",
       "       organism_ch1            series_id  \\\n",
       "0      Homo sapiens    GSE12781,GSE12783   \n",
       "1      Homo sapiens    GSE12782,GSE12783   \n",
       "2      Homo sapiens    GSE12781,GSE12783   \n",
       "3      Homo sapiens    GSE12782,GSE12783   \n",
       "4      Homo sapiens             GSE13008   \n",
       "...             ...                  ...   \n",
       "13162  Homo sapiens  GSE105937,GSE106146   \n",
       "13163  Homo sapiens  GSE105937,GSE106146   \n",
       "13164  Homo sapiens  GSE105937,GSE106146   \n",
       "13165  Homo sapiens  GSE105937,GSE106146   \n",
       "13166  Homo sapiens  GSE105937,GSE106146   \n",
       "\n",
       "                                         source_name_ch1  \\\n",
       "0                            Chromatin IP against Pol II   \n",
       "1      Chromatin IP against STAT1 in IFN gamma stimul...   \n",
       "2                            Chromatin IP against Pol II   \n",
       "3      Chromatin IP against STAT1 in IFN gamma stimul...   \n",
       "4                            Chromatin IP against Pol II   \n",
       "...                                                  ...   \n",
       "13162           p300 ChIP-seq from Senescent Fibroblasts   \n",
       "13163        IgG ChIP-seq from Proliferating Fibroblasts   \n",
       "13164        IgG ChIP-seq from Proliferating Fibroblasts   \n",
       "13165            IgG ChIP-seq from Senescent Fibroblasts   \n",
       "13166            IgG ChIP-seq from Senescent Fibroblasts   \n",
       "\n",
       "                                     characteristics_ch1 supplementary_file  \\\n",
       "0      HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...                NaN   \n",
       "1      HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...                NaN   \n",
       "2      HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...                NaN   \n",
       "3      HeLa S3 cells (ATCC/Snyder Lab);\\tgrowth prope...                NaN   \n",
       "4                                             K562 cells                NaN   \n",
       "...                                                  ...                ...   \n",
       "13162  cell line: IMR90;\\tcell type: Lung Fibroblasts...                NaN   \n",
       "13163  cell line: IMR90;\\tcell type: Lung Fibroblasts...                NaN   \n",
       "13164  cell line: IMR90;\\tcell type: Lung Fibroblasts...                NaN   \n",
       "13165  cell line: IMR90;\\tcell type: Lung Fibroblasts...                NaN   \n",
       "13166  cell line: IMR90;\\tcell type: Lung Fibroblasts...                NaN   \n",
       "\n",
       "      characteristics_ch2  \n",
       "0                     NaN  \n",
       "1                     NaN  \n",
       "2                     NaN  \n",
       "3                     NaN  \n",
       "4                     NaN  \n",
       "...                   ...  \n",
       "13162                 NaN  \n",
       "13163                 NaN  \n",
       "13164                 NaN  \n",
       "13165                 NaN  \n",
       "13166                 NaN  \n",
       "\n",
       "[13167 rows x 17 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#filter the table by chip-seq in gsm fields and then, group.by (gse) and count gsm\n",
    "\n",
    "df2_chipseq_title = df2[df2['title..9'].str.contains('chip-seq', case=False)].reset_index(drop=True)\n",
    "df2_chipseq_title \n",
    "                  \n",
    "#14387\n",
    "\n",
    "\n",
    "\n",
    "# df2_chipseq_grouped = df2_chipseq.groupby(['gse'])['gsm'].count().reset_index()\n",
    "# # df2.groupby(['gse'])['gsm'].count().reset_index()\n",
    "\n",
    "# df2_chipseq_grouped\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>type</th>\n",
       "      <th>title</th>\n",
       "      <th>summary</th>\n",
       "      <th>overall_design</th>\n",
       "      <th>status</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>gsm</th>\n",
       "      <th>title..9</th>\n",
       "      <th>type..10</th>\n",
       "      <th>gpl</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>series_id</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>characteristics_ch2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE16256</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project</td>\n",
       "      <td>The human embryonic stem cells (hESCs) are a u...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project...</td>\n",
       "      <td>Public on Aug 03 2009</td>\n",
       "      <td>19829295.0</td>\n",
       "      <td>GSM521926</td>\n",
       "      <td>Reference Epigenome: ChIP-Seq Input from IMR90...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE16256</td>\n",
       "      <td>IMR90 Cell Line, biological replicate 1 (r1); ...</td>\n",
       "      <td>molecule: genomic DNA;\\tdisease: None;\\tbiomat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE16256</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project</td>\n",
       "      <td>The human embryonic stem cells (hESCs) are a u...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project...</td>\n",
       "      <td>Public on Aug 03 2009</td>\n",
       "      <td>19829295.0</td>\n",
       "      <td>GSM521927</td>\n",
       "      <td>Reference Epigenome: ChIP-Seq Input from IMR90...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE16256</td>\n",
       "      <td>IMR90 Cell Line, biological replicate 1 (r1); ...</td>\n",
       "      <td>molecule: genomic DNA;\\tdisease: None;\\tbiomat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE16256</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project</td>\n",
       "      <td>The human embryonic stem cells (hESCs) are a u...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project...</td>\n",
       "      <td>Public on Aug 03 2009</td>\n",
       "      <td>19829295.0</td>\n",
       "      <td>GSM521928</td>\n",
       "      <td>Reference Epigenome: ChIP-Seq Input from IMR90...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE16256</td>\n",
       "      <td>IMR90 Cell Line, biological replicate 2; renla...</td>\n",
       "      <td>molecule: genomic DNA;\\tdisease: None;\\tbiomat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE16256</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project</td>\n",
       "      <td>The human embryonic stem cells (hESCs) are a u...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project...</td>\n",
       "      <td>Public on Aug 03 2009</td>\n",
       "      <td>19829295.0</td>\n",
       "      <td>GSM521929</td>\n",
       "      <td>Reference Epigenome: ChIP-Seq Input from IMR90...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE16256</td>\n",
       "      <td>IMR90 Cell Line, biological replicate 2; renla...</td>\n",
       "      <td>molecule: genomic DNA;\\tdisease: None;\\tbiomat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE16256</td>\n",
       "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project</td>\n",
       "      <td>The human embryonic stem cells (hESCs) are a u...</td>\n",
       "      <td>UCSD Human Reference Epigenome Mapping Project...</td>\n",
       "      <td>Public on Aug 03 2009</td>\n",
       "      <td>19829295.0</td>\n",
       "      <td>GSM521930</td>\n",
       "      <td>Reference Epigenome: ChIP-Seq Input from IMR90...</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL9115</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE16256</td>\n",
       "      <td>IMR90 Cell Line, biological replicate 2; renla...</td>\n",
       "      <td>molecule: genomic DNA;\\tdisease: None;\\tbiomat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1648</th>\n",
       "      <td>GSE81585</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>Transcriptome signatures of human induced plur...</td>\n",
       "      <td>We profiled the transcriptome of cardiomyocyte...</td>\n",
       "      <td>Cardiomyocytes derived from human induced plur...</td>\n",
       "      <td>Public on Nov 28 2018</td>\n",
       "      <td>30464173.0</td>\n",
       "      <td>GSM2183993</td>\n",
       "      <td>NR2F2</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE81585</td>\n",
       "      <td>fibroblasts</td>\n",
       "      <td>cell type: hiPSCs-CMs;\\tStage: day 30;\\treplic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1649</th>\n",
       "      <td>GSE117701</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>The mRNA export receptor NXF1 coordinates tran...</td>\n",
       "      <td>Alternative polyadenylation (APA) produces mRN...</td>\n",
       "      <td>In total 9 3'READS libraries for analysis of A...</td>\n",
       "      <td>Public on Jan 14 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM3307394</td>\n",
       "      <td>siCntl_input_rep1_ChIPseq</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE117701</td>\n",
       "      <td>cervical cancer cells</td>\n",
       "      <td>cell line: HeLa;\\ttreatment: Untreated;\\tproto...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1650</th>\n",
       "      <td>GSE117701</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>The mRNA export receptor NXF1 coordinates tran...</td>\n",
       "      <td>Alternative polyadenylation (APA) produces mRN...</td>\n",
       "      <td>In total 9 3'READS libraries for analysis of A...</td>\n",
       "      <td>Public on Jan 14 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM3307395</td>\n",
       "      <td>siCntl_pol2_IP_rep1_ChIPseq</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE117701</td>\n",
       "      <td>cervical cancer cells</td>\n",
       "      <td>cell line: HeLa;\\ttreatment: Untreated;\\tproto...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1651</th>\n",
       "      <td>GSE117701</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>The mRNA export receptor NXF1 coordinates tran...</td>\n",
       "      <td>Alternative polyadenylation (APA) produces mRN...</td>\n",
       "      <td>In total 9 3'READS libraries for analysis of A...</td>\n",
       "      <td>Public on Jan 14 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM3307396</td>\n",
       "      <td>siNXF1_input_rep1_ChIPseq</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE117701</td>\n",
       "      <td>cervical cancer cells</td>\n",
       "      <td>cell line: HeLa;\\ttreatment: siNXF1;\\tprotocol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1652</th>\n",
       "      <td>GSE117701</td>\n",
       "      <td>Expression profiling by high throughput sequen...</td>\n",
       "      <td>The mRNA export receptor NXF1 coordinates tran...</td>\n",
       "      <td>Alternative polyadenylation (APA) produces mRN...</td>\n",
       "      <td>In total 9 3'READS libraries for analysis of A...</td>\n",
       "      <td>Public on Jan 14 2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM3307397</td>\n",
       "      <td>siNXF1_pol2_IP_rep1_ChIPseq</td>\n",
       "      <td>SRA</td>\n",
       "      <td>GPL11154</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>GSE117701</td>\n",
       "      <td>cervical cancer cells</td>\n",
       "      <td>cell line: HeLa;\\ttreatment: siNXF1;\\tprotocol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1653 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gse                                               type  \\\n",
       "0      GSE16256  Genome binding/occupancy profiling by high thr...   \n",
       "1      GSE16256  Genome binding/occupancy profiling by high thr...   \n",
       "2      GSE16256  Genome binding/occupancy profiling by high thr...   \n",
       "3      GSE16256  Genome binding/occupancy profiling by high thr...   \n",
       "4      GSE16256  Genome binding/occupancy profiling by high thr...   \n",
       "...         ...                                                ...   \n",
       "1648   GSE81585  Expression profiling by high throughput sequen...   \n",
       "1649  GSE117701  Expression profiling by high throughput sequen...   \n",
       "1650  GSE117701  Expression profiling by high throughput sequen...   \n",
       "1651  GSE117701  Expression profiling by high throughput sequen...   \n",
       "1652  GSE117701  Expression profiling by high throughput sequen...   \n",
       "\n",
       "                                                  title  \\\n",
       "0        UCSD Human Reference Epigenome Mapping Project   \n",
       "1        UCSD Human Reference Epigenome Mapping Project   \n",
       "2        UCSD Human Reference Epigenome Mapping Project   \n",
       "3        UCSD Human Reference Epigenome Mapping Project   \n",
       "4        UCSD Human Reference Epigenome Mapping Project   \n",
       "...                                                 ...   \n",
       "1648  Transcriptome signatures of human induced plur...   \n",
       "1649  The mRNA export receptor NXF1 coordinates tran...   \n",
       "1650  The mRNA export receptor NXF1 coordinates tran...   \n",
       "1651  The mRNA export receptor NXF1 coordinates tran...   \n",
       "1652  The mRNA export receptor NXF1 coordinates tran...   \n",
       "\n",
       "                                                summary  \\\n",
       "0     The human embryonic stem cells (hESCs) are a u...   \n",
       "1     The human embryonic stem cells (hESCs) are a u...   \n",
       "2     The human embryonic stem cells (hESCs) are a u...   \n",
       "3     The human embryonic stem cells (hESCs) are a u...   \n",
       "4     The human embryonic stem cells (hESCs) are a u...   \n",
       "...                                                 ...   \n",
       "1648  We profiled the transcriptome of cardiomyocyte...   \n",
       "1649  Alternative polyadenylation (APA) produces mRN...   \n",
       "1650  Alternative polyadenylation (APA) produces mRN...   \n",
       "1651  Alternative polyadenylation (APA) produces mRN...   \n",
       "1652  Alternative polyadenylation (APA) produces mRN...   \n",
       "\n",
       "                                         overall_design  \\\n",
       "0     UCSD Human Reference Epigenome Mapping Project...   \n",
       "1     UCSD Human Reference Epigenome Mapping Project...   \n",
       "2     UCSD Human Reference Epigenome Mapping Project...   \n",
       "3     UCSD Human Reference Epigenome Mapping Project...   \n",
       "4     UCSD Human Reference Epigenome Mapping Project...   \n",
       "...                                                 ...   \n",
       "1648  Cardiomyocytes derived from human induced plur...   \n",
       "1649  In total 9 3'READS libraries for analysis of A...   \n",
       "1650  In total 9 3'READS libraries for analysis of A...   \n",
       "1651  In total 9 3'READS libraries for analysis of A...   \n",
       "1652  In total 9 3'READS libraries for analysis of A...   \n",
       "\n",
       "                     status   pubmed_id         gsm  \\\n",
       "0     Public on Aug 03 2009  19829295.0   GSM521926   \n",
       "1     Public on Aug 03 2009  19829295.0   GSM521927   \n",
       "2     Public on Aug 03 2009  19829295.0   GSM521928   \n",
       "3     Public on Aug 03 2009  19829295.0   GSM521929   \n",
       "4     Public on Aug 03 2009  19829295.0   GSM521930   \n",
       "...                     ...         ...         ...   \n",
       "1648  Public on Nov 28 2018  30464173.0  GSM2183993   \n",
       "1649  Public on Jan 14 2019         NaN  GSM3307394   \n",
       "1650  Public on Jan 14 2019         NaN  GSM3307395   \n",
       "1651  Public on Jan 14 2019         NaN  GSM3307396   \n",
       "1652  Public on Jan 14 2019         NaN  GSM3307397   \n",
       "\n",
       "                                               title..9 type..10       gpl  \\\n",
       "0     Reference Epigenome: ChIP-Seq Input from IMR90...      SRA   GPL9115   \n",
       "1     Reference Epigenome: ChIP-Seq Input from IMR90...      SRA   GPL9115   \n",
       "2     Reference Epigenome: ChIP-Seq Input from IMR90...      SRA   GPL9115   \n",
       "3     Reference Epigenome: ChIP-Seq Input from IMR90...      SRA   GPL9115   \n",
       "4     Reference Epigenome: ChIP-Seq Input from IMR90...      SRA   GPL9115   \n",
       "...                                                 ...      ...       ...   \n",
       "1648                                              NR2F2      SRA  GPL11154   \n",
       "1649                          siCntl_input_rep1_ChIPseq      SRA  GPL11154   \n",
       "1650                        siCntl_pol2_IP_rep1_ChIPseq      SRA  GPL11154   \n",
       "1651                          siNXF1_input_rep1_ChIPseq      SRA  GPL11154   \n",
       "1652                        siNXF1_pol2_IP_rep1_ChIPseq      SRA  GPL11154   \n",
       "\n",
       "      organism_ch1  series_id  \\\n",
       "0     Homo sapiens   GSE16256   \n",
       "1     Homo sapiens   GSE16256   \n",
       "2     Homo sapiens   GSE16256   \n",
       "3     Homo sapiens   GSE16256   \n",
       "4     Homo sapiens   GSE16256   \n",
       "...            ...        ...   \n",
       "1648  Homo sapiens   GSE81585   \n",
       "1649  Homo sapiens  GSE117701   \n",
       "1650  Homo sapiens  GSE117701   \n",
       "1651  Homo sapiens  GSE117701   \n",
       "1652  Homo sapiens  GSE117701   \n",
       "\n",
       "                                        source_name_ch1  \\\n",
       "0     IMR90 Cell Line, biological replicate 1 (r1); ...   \n",
       "1     IMR90 Cell Line, biological replicate 1 (r1); ...   \n",
       "2     IMR90 Cell Line, biological replicate 2; renla...   \n",
       "3     IMR90 Cell Line, biological replicate 2; renla...   \n",
       "4     IMR90 Cell Line, biological replicate 2; renla...   \n",
       "...                                                 ...   \n",
       "1648                                        fibroblasts   \n",
       "1649                              cervical cancer cells   \n",
       "1650                              cervical cancer cells   \n",
       "1651                              cervical cancer cells   \n",
       "1652                              cervical cancer cells   \n",
       "\n",
       "                                    characteristics_ch1 supplementary_file  \\\n",
       "0     molecule: genomic DNA;\\tdisease: None;\\tbiomat...                NaN   \n",
       "1     molecule: genomic DNA;\\tdisease: None;\\tbiomat...                NaN   \n",
       "2     molecule: genomic DNA;\\tdisease: None;\\tbiomat...                NaN   \n",
       "3     molecule: genomic DNA;\\tdisease: None;\\tbiomat...                NaN   \n",
       "4     molecule: genomic DNA;\\tdisease: None;\\tbiomat...                NaN   \n",
       "...                                                 ...                ...   \n",
       "1648  cell type: hiPSCs-CMs;\\tStage: day 30;\\treplic...                NaN   \n",
       "1649  cell line: HeLa;\\ttreatment: Untreated;\\tproto...                NaN   \n",
       "1650  cell line: HeLa;\\ttreatment: Untreated;\\tproto...                NaN   \n",
       "1651  cell line: HeLa;\\ttreatment: siNXF1;\\tprotocol...                NaN   \n",
       "1652  cell line: HeLa;\\ttreatment: siNXF1;\\tprotocol...                NaN   \n",
       "\n",
       "     characteristics_ch2  \n",
       "0                    NaN  \n",
       "1                    NaN  \n",
       "2                    NaN  \n",
       "3                    NaN  \n",
       "4                    NaN  \n",
       "...                  ...  \n",
       "1648                 NaN  \n",
       "1649                 NaN  \n",
       "1650                 NaN  \n",
       "1651                 NaN  \n",
       "1652                 NaN  \n",
       "\n",
       "[1653 rows x 17 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2_char_chip = df2[df2['characteristics_ch1'].str.contains('chip-seq', case=False, na=False)].reset_index(drop=True)\n",
    "df2_char_chip    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>gsm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE100040</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE100099</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE100306</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE100347</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE100375</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3026</th>\n",
       "      <td>GSE99626</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3027</th>\n",
       "      <td>GSE99680</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3028</th>\n",
       "      <td>GSE99943</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3029</th>\n",
       "      <td>GSE99950</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3030</th>\n",
       "      <td>GSE99998</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3031 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gse  gsm\n",
       "0     GSE100040   28\n",
       "1     GSE100099   11\n",
       "2     GSE100306    4\n",
       "3     GSE100347    6\n",
       "4     GSE100375    2\n",
       "...         ...  ...\n",
       "3026   GSE99626   21\n",
       "3027   GSE99680   78\n",
       "3028   GSE99943    1\n",
       "3029   GSE99950    1\n",
       "3030   GSE99998    4\n",
       "\n",
       "[3031 rows x 2 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2_chipseq_total = pd.concat([df2_chipseq_title, df2_char_chip], ignore_index=True)\n",
    "\n",
    "# df2_chipseq_total\n",
    "\n",
    "\n",
    "df2_chipseq_total_grouped = df2_chipseq_total.groupby(['gse'])['gsm'].count().reset_index()\n",
    "df2_chipseq_total_grouped\n",
    "\n",
    "#when we filter to chip-seq data by gsm fields, we lost 4721 - 3031 = 1690 GSEs \n",
    "#the first one (28 GMS) match with NCBI\n",
    "#the second one = the correct number is 15, because there are 4 GSM that just contains ChIP in title. However, the \n",
    "##library strategy has the ChIP-Seq information. But the geometadb dont have this information..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[['GSE10437', 'Samples (4)']]\n"
     ]
    }
   ],
   "source": [
    "import urllib.request\n",
    "import lxml.html as lh\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import re\n",
    "from time import sleep\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "regex = re.compile('Samples\\s+\\(\\d+\\)')  #get the sample (number)\n",
    "gse_samples_ncbi =[]\n",
    "\n",
    "for gse in gse_list_GEOdb:\n",
    "    url = 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=' + gse\n",
    "    response = requests.get(url)\n",
    "    data = response.text\n",
    "    soup = BeautifulSoup(data, 'lxml')\n",
    "    tr = soup.findAll(\"tr\", {\"valign\" : \"top\"}) #ok    \n",
    "\n",
    "    for td in tr:\n",
    "        match = re.search(regex, td.text)\n",
    "        if match:\n",
    "            gse_samples_ncbi.append([gse, match.group(0)])\n",
    "    \n",
    "    sleep(3) #Ok\n",
    "\n",
    "print(gse_samples_ncbi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "148\n"
     ]
    }
   ],
   "source": [
    "print(len(gse_samples_ncbi))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filtering the gse_list_GEOdb to dont lose just the list with 2181 GSEs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2181\n"
     ]
    }
   ],
   "source": [
    "#converting list of list in a simple list\n",
    "\n",
    "# simple_list_gse_ncbi_breaked = [item for sublist in gse_samples_ncbi for item in sublist]\n",
    "\n",
    "# simple_list_gse_ncbi_breaked\n",
    "\n",
    "\n",
    "\n",
    "# list_gse_ncbi_breaked = []\n",
    "\n",
    "# for i in simple_list_gse_ncbi_breaked:\n",
    "#     if 'GSE' in i:\n",
    "#         list_gse_ncbi_breaked.append(i)\n",
    "        \n",
    "# print(len(list_gse_ncbi_breaked))   \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2540\n"
     ]
    }
   ],
   "source": [
    "# list_difference = set(gse_list_GEOdb).difference(list_gse_ncbi_breaked) \n",
    "\n",
    "# print(len(list_difference)) #ok"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "ename": "SSLError",
     "evalue": "HTTPSConnectionPool(host='www.ncbi.nlm.nih.gov', port=443): Max retries exceeded with url: /geo/query/acc.cgi?acc=GSE93975 (Caused by SSLError(SSLError(\"bad handshake: SysCallError(60, 'ETIMEDOUT')\")))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSysCallError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/contrib/pyopenssl.py\u001b[0m in \u001b[0;36mwrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname)\u001b[0m\n\u001b[1;32m    484\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 485\u001b[0;31m                 \u001b[0mcnx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    486\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOpenSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mWantReadError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/OpenSSL/SSL.py\u001b[0m in \u001b[0;36mdo_handshake\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1933\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_lib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL_do_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_ssl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1934\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_ssl_error\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_ssl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1935\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/OpenSSL/SSL.py\u001b[0m in \u001b[0;36m_raise_ssl_error\u001b[0;34m(self, ssl, result)\u001b[0m\n\u001b[1;32m   1662\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0merrno\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1663\u001b[0;31m                         \u001b[0;32mraise\u001b[0m \u001b[0mSysCallError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merrorcode\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1664\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mSysCallError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"Unexpected EOF\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSysCallError\u001b[0m: (60, 'ETIMEDOUT')",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    671\u001b[0m                 \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 672\u001b[0;31m                 \u001b[0mchunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunked\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    673\u001b[0m             )\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    375\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 376\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_conn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    377\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m    993\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"sock\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# AppEngine might not have  `.sock`\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 994\u001b[0;31m             \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    995\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/connection.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    393\u001b[0m             \u001b[0mserver_hostname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mserver_hostname\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 394\u001b[0;31m             \u001b[0mssl_context\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcontext\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    395\u001b[0m         )\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/util/ssl_.py\u001b[0m in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password)\u001b[0m\n\u001b[1;32m    369\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mHAS_SNI\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mserver_hostname\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 370\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mcontext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrap_socket\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mserver_hostname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mserver_hostname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/contrib/pyopenssl.py\u001b[0m in \u001b[0;36mwrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname)\u001b[0m\n\u001b[1;32m    490\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOpenSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 491\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mssl\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSLError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"bad handshake: %r\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    492\u001b[0m             \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: (\"bad handshake: SysCallError(60, 'ETIMEDOUT')\",)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    448\u001b[0m                     \u001b[0mretries\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax_retries\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 449\u001b[0;31m                     \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    450\u001b[0m                 )\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    719\u001b[0m             retries = retries.increment(\n\u001b[0;32m--> 720\u001b[0;31m                 \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_pool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_stacktrace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    721\u001b[0m             )\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/urllib3/util/retry.py\u001b[0m in \u001b[0;36mincrement\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    435\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnew_retry\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_exhausted\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 436\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMaxRetryError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_pool\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mResponseError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcause\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='www.ncbi.nlm.nih.gov', port=443): Max retries exceeded with url: /geo/query/acc.cgi?acc=GSE93975 (Caused by SSLError(SSLError(\"bad handshake: SysCallError(60, 'ETIMEDOUT')\")))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-35ac1e19c0a8>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mgse\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlist_difference\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0murl\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc='\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mgse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0msoup\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'lxml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     73\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     74\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 75\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     76\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    531\u001b[0m         }\n\u001b[1;32m    532\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 533\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    534\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    535\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    644\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    645\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 646\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    647\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    648\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/lab-jacques/lib/python3.7/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    512\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreason\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_SSLError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    513\u001b[0m                 \u001b[0;31m# This branch is for urllib3 v1.22 and later.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 514\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mSSLError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    515\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    516\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='www.ncbi.nlm.nih.gov', port=443): Max retries exceeded with url: /geo/query/acc.cgi?acc=GSE93975 (Caused by SSLError(SSLError(\"bad handshake: SysCallError(60, 'ETIMEDOUT')\")))"
     ]
    }
   ],
   "source": [
    "# #Run the code with absence GSE in the first downloaded list\n",
    "\n",
    "\n",
    "# gse_samples_ncbi_2 =[]\n",
    "\n",
    "# for gse in list_difference:\n",
    "#     url = 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=' + gse\n",
    "#     response = requests.get(url)\n",
    "#     data = response.text\n",
    "#     soup = BeautifulSoup(data, 'lxml')\n",
    "#     tr = soup.findAll(\"tr\", {\"valign\" : \"top\"}) #ok    \n",
    "\n",
    "#     for td in tr:\n",
    "#         match = re.search(regex, td.text)\n",
    "#         if match:\n",
    "#             gse_samples_ncbi.append([gse, match.group(0)])\n",
    "    \n",
    "#     sleep(2) #Ok\n",
    "\n",
    "# print(gse_samples_ncbi)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# V2\n",
    "import urllib.request\n",
    "import lxml.html as lh\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import re\n",
    "from time import sleep\n",
    "\n",
    "\n",
    "gse_list_samples_file = open(\"gse_list_sample_file.tsv\", \"a+\")\n",
    "\n",
    "regex = re.compile('Samples\\s+\\(\\d+\\)')  #get the sample (number)\n",
    "\n",
    "err_samp = []\n",
    "index_new_start = 2484\n",
    "line = ''\n",
    "for i,gse in enumerate(gse_list_GEOdb[index_new_start:]):\n",
    "    try:\n",
    "        url = 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=' + gse\n",
    "        response = requests.get(url)\n",
    "        data = response.text\n",
    "        soup = BeautifulSoup(data, 'lxml')\n",
    "        tr = soup.findAll(\"tr\", {\"valign\" : \"top\"}) #ok    \n",
    "    except:\n",
    "        print('Error on sample', gse)\n",
    "        err_samp.append(gse)          # Get problematic samples (try later)\n",
    "        sleep(60)                     # sleep more so NCBI let me pass\n",
    "        \n",
    "    else:\n",
    "        for td in tr:\n",
    "            match = re.search(regex, td.text)\n",
    "            if match:\n",
    "                line = str(i + index_new_start) +\"\\t\"+ gse +\"\\t\"+ match.group(0) +\"\\n\"\n",
    "                gse_list_samples_file.write(line)\n",
    "    \n",
    "    sleep(4) #Ok\n",
    "\n",
    "# print(gse_samples_ncbi)\n",
    "gse_list_samples_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse_list_samples_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GSE13313', 'Samples (16)']\n"
     ]
    }
   ],
   "source": [
    "print(gse_samples_ncbi[10])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "print(len(err_samp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse_list_sample_file = open(\"gse_list_sample_file.tsv\", \"w\")\n",
    "for i,j in enumerate(gse_samples_ncbi):\n",
    "    line = str(i) + \"\\t\" + j[0] + \"\\t\" + j[1] + \"\\n\"\n",
    "    gse_list_sample_file.write(line)\n",
    "\n",
    "gse_list_sample_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2483\n"
     ]
    }
   ],
   "source": [
    "print(gse_list_GEOdb.index('GSE95980'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSE95971\n"
     ]
    }
   ],
   "source": [
    "print(gse_list_GEOdb[2474])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
